migrations/20220331202635-create-journals.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    await queryInterface.sequelize.transaction(async (transaction) => {
+      await queryInterface.createTable(
+        'journals',
+        {
+          id: {
+            type: Sequelize.UUID,
+            primaryKey: true,
+            allowNull: false,
+            unique: true,
+            autoIncrement: false,
+            defaultValue: Sequelize.fn('uuid_generate_v4'),
+          },
+          action: {
+            type: Sequelize.ENUM,
+            allowNull: false,
+            values: [
+              'login',
+              'get_places',
+              'edit_place',
+              'add_comment',
+              'edit_comment',
+              'delete_comment',
+              'add_demand',
+              'edit_demand',
+              'delete_demand',
+              'delete_all_demands',
+            ],
+          },
+          details: {
+            type: Sequelize.STRING,
+            allowNull: true,
+            defaultValue: null,
+          },
+          user_id: {
```

Comment:
I would suggest not to bind journaling table with user (or any other entity), but instead - save user login in `user` string field. Then, even if the user is completely deleted from database, journal will persist and we can tell that i.e. user `X` has deleted a `place` on April 1, even if user `X` is no longer in db. Binding user with journal by FK will completely erase the history in above scenario.

Reply:
Sure, makes a lot of sense